### PR TITLE
Copy in release build workflow from opentelemetry-java

### PR DESCRIPTION
**Pull request by anuraaga**
Almost the same as opentelemetry-java except without README update steps.

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -13,22 +13,69 @@ This signals Nebula plugin to build and publish to
...
-That build, after usual building and checking, will run `./gradlew -Prelease.scope=patch final`.
-This will signal Nebula plugin to build a new version `vX.Y.(Z+1)` and publish it to Bintray repo.
+## Starting the Release
+
+Open the release build workflow in your browser [here](https://github.com/open-telemetry/opentelemetry-java-instrumentation/actions?query=workflow%3A%22Release+Build%22).
+
+You will see a button that says "Run workflow". Press the button, enter the version number you want
+to release in the input field that pops up, and then press "Run workflow".
+
+This triggers the release process, which builds the artifacts, updates the README with the new
```
Comment (iNikem) [inline RELEASING.md:23]: We don't update README in this repo, right?

File: .github/workflows/release-build.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        required: true
+
+jobs:
+  build:
+    name: Build and release
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
```
Comment (iNikem) [inline .github/workflows/release-build.yml:17]: This repo does not have submodules

File: .github/workflows/release-build.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        with:
+          job-id: jdk11
+          remote-build-cache-proxy-enabled: false
+          arguments: build --stacktrace -Prelease.version=${{ github.event.inputs.version }}
+      - name: Publish artifacts
+        run: ./gradlew final --stacktrace -Prelease.version=${{ github.event.inputs.version }}
+        env:
+          BINTRAY_USER: ${{ secrets.BINTRAY_USER }}
+          BINTRAY_KEY: ${{ secrets.BINTRAY_KEY }}
+          GRGIT_USER: ${{ github.actor }}
```
Comment (iNikem) [inline .github/workflows/release-build.yml:31]: We should not need `GRGIT` or?

File: .github/workflows/patch-release-build.yml [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+              # Trim whitespaces and cherrypick
+              echo $word | sed 's/ *$//g' | sed 's/^ *//g' | git cherry-pick --stdin
+          done
+      - uses: burrunan/gradle-cache-action@v1.5
+        with:
+          job-id: jdk11
+          remote-build-cache-proxy-enabled: false
+          arguments: build --stacktrace -Prelease.version=${{ github.event.inputs.version }}
+          properties: |
+            enable.docker.tests=true
```
Comment (iNikem) [inline .github/workflows/patch-release-build.yml:74]: We don't need this, do we?

File: .github/workflows/patch-release-build.yml [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+      - run: git push
+      - name: Checkout master
+        uses: actions/checkout@v2
+        with:
+           submodules: true
+      - uses: burrunan/gradle-cache-action@v1.5
+        with:
+          job-id: jdk11
+          remote-build-cache-proxy-enabled: false
+          arguments: --stacktrace -Prelease.version=${{ github.event.inputs.version }}
```
Comment (iNikem) [inline .github/workflows/patch-release-build.yml:95]: What does this do?

File: .github/workflows/release-build.yml (hunk 3 above)
Comment (anuraaga) [reply .github/workflows/release-build.yml:29]: nebula will create and push a tag so GRGIT credentials allows it to do that.

**Review by anuraaga — COMMENTED** [review]
Thanks

File: .github/workflows/release-build.yml (hunk 3 above)
Comment (iNikem) [reply .github/workflows/release-build.yml:29]: But `prepare-release-branch` job above creates and pushes branch too. And without GRGIT?

Comment (anuraaga) [reply .github/workflows/release-build.yml:29]: Oops forgot to respond - yeah normal `git` commands authenticate automatically. I first tried without GRGIT hoping it'd behave the same in Gradle but it doesn't :(

**Review by iNikem — APPROVED** [review]

**Pull request merged**